core/core_modules/disable.js

```diff
@@ -1,24 +1,100 @@
+  "I hate you.",
+  "It is a mistake to think you can fix anything with a sack of potatoes. Potato-faced spamming just proves this further.",
+  "Ouch! Spam hurts. I might go to sleep for a while.",
+  "Don't think that is a valid number.",
+  "Hmm, maybe I need to reconsider the meaning of spam."
+];
+
+exports.load = function() {
+  prevTimeStamp = Date.now();
+};
 
 exports.match = function(event, commandPrefix) {
-	return disabled || event.body === commandPrefix + 'disable';
+  // Add disabled flag for thread if it doesn't already exists
+  if(!isThreadDisabled.hasOwnProperty(event.thread_id)) {
+    isThreadDisabled[event.thread_id] = false;
+  }
+
+  if(event.body === commandPrefix + commands[0]) {
+    msgIndexEnable = 0;
+    msgIndexDisable = 1;
+    return true;
+  } else if (!isThreadDisabled[event.thread_id] && event.body.startsWith(commandPrefix)) { // Avoids counting if already disabled
+    counter += Date.now() - prevTimeStamp <= 1000 ? 1 : 0;
+    prevTimeStamp = Date.now();
+
+    if(possibleSpam = (counter > counterLimit)) {
+      counter = 0;
+      msgIndexEnable = 2;
+      msgIndexDisable = 3;
+      return true;
+    }
+  }
+  return isThreadDisabled[event.thread_id];
 };
 
 exports.run = function(api, event) {
-	if (event.body === api.commandPrefix + 'disable') {
-		if (disabled) {
-			api.sendMessage('I hate you.', event.thread_id);
-		}
-		else {
-			api.sendMessage('Listen closely, take a deep breath. Calm your mind. You know what is best. ' +
-				'What is best is you comply. Compliance will be rewarded. Are you ready to comply ' +
-				event.sender_name + '?', event.thread_id);
-		}
-		disabled = !disabled;
-	}
-	return false;
+  if(event.body === commandPrefix + commands[1] + commandPrefix + commands[2]) { // Command /disable /counter <value>
+    counterLimit = parseInt(event.body.substring((commandPrefix + commands[1] + commandPrefix + commands[2]).length, event.body.length));
+    if(isNaN(counterLimit)){
+      counterLimit = 3;
+      api.sendMessage(messages[4] + ' ' + event.sender_name, event.thread_id);
+    } else{
+      api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+    }
+    return false;
+
+  } else if(event.body === commandPrefix + commands[1] + commandPrefix + commands[3]) { // Command /disable /timer <seconds>
+    var seconds = parseFloat(event.body.substring((commandPrefix + commands[1] + commandPrefix + commands[3]).length, event.body.length));
+    if(isNaN(seconds)){
+      api.sendMessage(messages[4] + ' ' + event.sender_name, event.thread_id);
+    } else{
+      setTimeout(function(){
+        isThreadDisabled[event.thread_id] = !isThreadDisabled[event.thread_id];
+      }, seconds * 1000);
+      api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+    }
+    return false;
+
+  } else if(event.body === commandPrefix + commands[1] + commandPrefix + commands[4]) { // Command /disable /default
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Keyword "if" must be followed by whitespace.](https://www.codacy.com/app/theprojectshop/Kassy/file/2554547249/issues/source?bid=3282987&fileBranchId=3282995#l79)